src/sensors/mod.rs

```diff
@@ -0,0 +1,81 @@
+//! Module defining common sensor functionality.
+
+mod humidity;
+mod people_now_present;
+mod power_consumption;
+mod temperature;
+
+pub use humidity::{HumiditySensor, HumiditySensorTemplate};
+pub use people_now_present::{PeopleNowPresentSensor, PeopleNowPresentSensorTemplate};
+pub use power_consumption::{PowerConsumptionSensor, PowerConsumptionSensorTemplate};
+pub use temperature::{TemperatureSensor, TemperatureSensorTemplate};
+
+use log::warn;
+use serde::{Deserialize, Serialize};
+
+/// Common information describing any sensor.
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct SensorMetadata {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub location: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+}
+
+/// Common information describing any sensor which requires a specified location.
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct LocalisedSensorMetadata {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    pub location: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+}
+
+/// A trait for all possible sensor templates.
+///
+/// A sensor template is like a sensor struct, but without the actual data in it.
+/// A `SensorTemplate` is capable of registering itself in a `Sensors` struct.
+pub trait SensorTemplate: Send + Sync {
+    fn to_sensor(&self, value_str: &str, sensors: &mut Sensors) {
+        if let Err(e) = self.try_to_sensor(value_str, sensors) {
+            warn!("Omitting sensor. Reason: {}", e);
+        }
+    }
+
+    fn try_to_sensor(&self, value_str: &str, sensors: &mut Sensors)
+        -> Result<(), Box<dyn std::error::Error>>;
```

Comment:
I'm not sure if we'd prefer a custom error type (since currently the only possible error is a parsing error as far as I can tell) instead of `Box<dyn Error>`, but we can also refactor it at a later stage.

Reply:
Yes, a custom error would be cleaner. I think I just wanted to get something functional to review.

I would like to address this before this PR is merged.

Reply:
Using `thiserror` is the better approach 👍🏻